index.js

```diff
@@ -56,12 +56,12 @@ function interpolate(t, degree, points, knots, weights, result) {
     v[i][d] = weights[i];
   }
 
-  // l (level) goes from 1 to the curve degree
+  // l (level) goes from 1 to the curve degree + 1
   var alpha;
-  for(l=1; l<=degree; l++) {
+  for(l=1; l<=degree+1; l++) {
     // build level l of the pyramid
-    for(i=s; i>s-degree+l; i--) {
-      alpha = (t - knots[i]) / (knots[i+degree-l] - knots[i]);
+    for(i=s; i>s-degree-1+l; i--) {
```

Comment:
since the terminology is now "degree" I'll make a small update to the test to make it pass "degree" around as well (it still uses "order" at the moment)